hbase-server/src/main/java/org/apache/hadoop/hbase/master/MetaFixer.java

```diff
@@ -242,17 +243,35 @@ void fixOverlaps(CatalogJanitor.Report report) throws IOException {
     }
     List<SortedSet<RegionInfo>> merges = new ArrayList<>();
     SortedSet<RegionInfo> currentMergeSet = new TreeSet<>();
+    HashSet<RegionInfo> regionsInMergeSet = new HashSet<>();
     RegionInfo regionInfoWithlargestEndKey =  null;
     for (Pair<RegionInfo, RegionInfo> pair: overlaps) {
       if (regionInfoWithlargestEndKey != null) {
         if (!isOverlap(regionInfoWithlargestEndKey, pair) ||
             currentMergeSet.size() >= maxMergeCount) {
-          merges.add(currentMergeSet);
-          currentMergeSet = new TreeSet<>();
+          if (currentMergeSet.size() <= 1) {
```

Comment:
This bit is hard to read. A comment on why <= 1?

Reply:
In the case of the merge set contains only 1 region or empty, we do not need to submit this merge request as no merge is going to happen. I will update the patch with comments to explain this case.

Reply:
added comments.